src/types/Message.ts

```diff
@@ -0,0 +1,200 @@
+  content: Content
+  type: Type
+}
+
+
+/**
+ * - `body` - The body of the message, see [[MessageBody]].
+ * - `createdAt` - The timestamp of the message construction.
+ * - `sender` - The DID of the sender.
+ * - `receiver` - The DID of the receiver.
+ * - `messageId` - The message id.
+ * - `receivedAt` - The timestamp of the message reception.
+ * - `inReplyTo` - The id of the parent-message.
+ * - `references` - The references or the in-reply-to of the parent-message followed by the message-id of the parent-message.
+ */
+export interface IMessage<Body extends IMessageBodyBase = {type: string, content: unknown}> {
+  body: Body
+  createdAt: number
+  sender: DidUri
+  receiver: DidUri
+  messageId?: string
+  receivedAt?: number
+  inReplyTo?: IMessage<Body>['messageId']
+  references?: Array<IMessage<Body>['messageId']>
+}
+
+/**
+ * Error messages signal unintentional programming errors which happened during
+ * the processing of the incoming messages or when constructing a response message.
+ */
+export type IError = IMessageBodyBase<'error', {name? : string, message?: string}>;
+
+/**
+ * Rejection messages signal the intentional cancelling of an individual step in the flow.
+ */
+export type IReject = IMessageBodyBase<'reject', {name? : string, message?: string}>;
+
+/**
+ * An attester utilizes the message to propose a claim. The purpose of the extension is to enable
+ * the user to authorize and endorse the claims prepared by the attester.
+ */
+export type ISubmitTerms = IMessageBodyBase<'submit-terms', ITerms>;
+
+
+/**
+ * The content of the [IRequestAttestation] message.
+ */
+
+export interface IRequestAttestationContent {
+  credential: ICredential
+  quote?: IQuoteAgreement
+}
+
+/**
+ * The extension only sends the request with active consent of the user. This is the first step
+ * where the user’s DID is revealed to the dApp.
+ */
+export type IRequestAttestation = IMessageBodyBase<'request-attestation', IRequestAttestationContent>
+
+
+/**
+ * The content of the [IRequestPayment] message.
+ */
+export interface IRequestPaymentContent {
+  // Same as the `rootHash` value of the `'request-attestation'` message */
+  claimHash: string
+}
+
+/**
+ * An attester can send this message if it wants the user to transfer payment in KILT Coins by themselves without interrupting the flow.
+ */
+export type IRequestPayment = IMessageBodyBase<'request-payment', IRequestPaymentContent>
+
+
+/**
+ * The content of the [ISubmitAttestation] message.
+ */
+export interface ISubmitAttestationContent {
+  attestation: IAttestation
+}
+
+/**
+ * The attester sends the valid credential to the extension.
+ */
+export type ISubmitAttestation = IMessageBodyBase<'submit-attestation', ISubmitAttestationContent>
+
+
+/**
+ * If the attester does not approve the attestation request, the extension receives the [IRejectAttestation] message.
+ */
+export type IRejectAttestation = IMessageBodyBase<'reject-attestation', ICredential['rootHash']>
+
+
```

Comment:
White space is a little off in this whole file. Normally you do two new lines between top level definitions.

Reply:
addressed in: [d2627c3](https://github.com/KILTprotocol/kilt-extension-api/pull/18/commits/d2627c3650aa91bf69830e8e877946fd17c12e29)